### PR TITLE
💥 feat: bump terraform to `1.4.4`

### PR DESCRIPTION
**Pull request by dduportal**
This PR introduces 2 changes:

- 💥 feat: bump terraform to ~`1.3.3`~ `1.4.4` (from `1.1.9`)
- Change the updatecli manifest tracking the Terraform version to always check the latest semver version. The goal is to avoid waiting too much for Terraform minor (or major) upgrades: all of our project have a guardrail with the Terraform version being constrained.

### CODE REVIEW

**Comment by dduportal** [comment]
Update: we need to switch to 1.4.4 as per https://github.com/jenkins-infra/packer-images/pull/591

File: cst.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ metadataTest:
     - key: io.jenkins-infra.tools
       value: "aws-cli,azure-cli,golang,golangci-lint,jenkins-inbound-agent,packer,terraform,tfsec,updatecli,yq"
     - key: io.jenkins-infra.tools.terraform.version
-      value: "1.1.9"
+      value: "1.3.3"
```
Comment (timja) [inline cst.yml:7]:
```suggestion
      value: "1.4.4"
```

**Review by timja — APPROVED** [review]

File: updatecli/updatecli.d/terraform.yml [hunk 2]
```diff
@@ -23,9 +23,7 @@ sources:
       token: "{{ requiredEnv .github.token }}"
       username: "{{ .github.username }}"
       versionfilter:
-        kind: regex
-        # Stick to versions 1.1.x of terraform
-        pattern: 'v1\.1\.(\d*)$'
+        kind: semver
```
Comment (dduportal) [inline updatecli/updatecli.d/terraform.yml:26]:
```suggestion
        kind: semver
        pattern: "~1.4"
```

**Comment by dduportal** [comment]
Please note that, once merged, we'll have to stop what we are doing and update all terraform projects one by one

**Comment by dduportal** [comment]
> > to pin the version until the transition is done.
> 
> Or just keep it there so this isn't a concern again 😄. Per project terraform versions work much better imo.

Yes, but it is also important that we keep Terrafom up to date and it's nice of you to point it out! I was only trying to write the "path" to allow better delegation and knowledge sharing

**Comment by lemeurherve** [comment]
Selected notes from changelogs:

https://github.com/hashicorp/terraform/blob/v1.2/CHANGELOG.md
> When showing a plan, Terraform CLI will now only show "Changes outside of Terraform" if they relate to resources and resource attributes that contributed to the changes Terraform is proposing to make.

https://github.com/hashicorp/terraform/blob/v1.3/CHANGELOG.md
> Optional attributes for object type constraints: When declaring an input variable whose type constraint includes an object type, you can now declare individual attributes as optional, and specify a default value to use if the caller doesn't set it.

> Added functions: startswith and endswith allow you to check whether a given string has a specified prefix or suffix. 

> terraform show -json: Output changes now include more detail about the unknown-ness of the planned value. Previously, a planned output would be marked as either fully known or partially unknown, with the after_unknown field having value false or true respectively. Now outputs correctly expose the full structure of unknownness for complex values, allowing consumers of the JSON output format to determine which values in a collection are known only after apply.

https://github.com/hashicorp/terraform/blob/v1.4/CHANGELOG.md
> terraform plan can now store a plan file even when encountering errors, which can later be inspected to help identify the source of the failures

**Review by lemeurherve — APPROVED** [review]
LGTM

**Comment by dduportal** [comment]
Moving back to draft until we can plan this change properly

**Comment by dduportal** [comment]
Closing as jenkins-infra/packer-images already has a recent Terraform

**Pull request closed without merging**